data/src/main/java/org/apache/iceberg/data/orc/GenericOrcWriters.java

```diff
@@ -326,7 +326,7 @@ public void nonNullWrite(int rowId, LocalDateTime data, ColumnVector output) {
     public void nonNullWrite(int rowId, BigDecimal data, ColumnVector output) {
       // TODO: validate precision and scale from schema
       ((DecimalColumnVector) output).vector[rowId]
-          .setFromLongAndScale(data.unscaledValue().longValueExact(), scale);
+          .setFromLongAndScale(data.unscaledValue().longValueExact(), data.scale());
```

Comment:
What about the TODO to check the scale matches the column's scale? As long as we're updating this, does it make sense to fix that, since we just had a decimal scale problem?